contracts/FlowServiceAccount.cdc

```diff
@@ -92,9 +97,21 @@ pub contract FlowServiceAccount {
 
     /// Returns true if the given address is permitted to create accounts, false otherwise
     pub fun isAccountCreator(_ address: Address): Bool {
+        // If account creation is not restricted, then anyone can create an account
+        if !self.isAccountCreationRestricted() {
+            return true
+        }
         return self.accountCreators[address] ?? false
     }
 
+    /// Is true if new acconts can only be created by approved accounts `self.accountCreators`
+    pub fun isAccountCreationRestricted(): Bool {
+        let value = self.account.load<Bool>(from: /storage/isAccountCreationRestricted) ?? false
+        self.account.save<Bool>(value ,to: /storage/isAccountCreationRestricted)
+        return value
```

Comment:
No need to remove the value from storage and write it back:

```suggestion
        return self.account.copy<Bool>(from: /storage/isAccountCreationRestricted) ?? false
```

Reply:
Ah! Did not know about the `copy` method. Must have missed it in the docs. Thank you for this.

Reply:
@turbolent Saving the value to storage if it was copied causes the save fail, at least when I have tried, so I have to load it during my testing.  Is this not how it is supposed to behave?

Reply:
Yes, this is the behaviour of save, i.e. it aborts if there is already a stored value.
Here, there is no need to first remove it from storage and then put it back, the function only reads it and the read value is a boolean, so it can just be copied. Note that the suggestion replaces the load + save with just a single copy

Reply:
oh yeah you're right. I didn't look close enough. My bad!